### PR TITLE
[sc-11899] Remove Adapter object from templates

### PR DESCRIPTION
**Pull request by nholbrook**
- Moves the `adapter.config` field to `adapterConfig`
- Moves the `adapter.auth` field to `auth`
- Adds a `type` field
- Removes the `adapter` object

This should not be merged until after the deployment on Aug 23rd. This can be merged anytime after in time for the release on Aug 30th. This should be deployed with [sc-11904](https://app.shortcut.com/sgnl/story/11904/fe-phase-4-update-sor-management-ui-to-support-adapter-resources)

### CODE REVIEW

**Comment by nholbrook** [comment]
cc @ivanfong for the format changes

File: azure-ad.sgnl.yaml [hunk 1]
```diff
@@ -8,19 +8,16 @@ defaultSyncFrequency: HOURLY
...
-  disableTls: true
-  auth:
-    - oAuth2ClientCredentials:
-        clientId: "{{Input Required: Client-ID}}"
-        scope: "https://graph.microsoft.com/.default"
-        tokenUrl: "https://login.microsoftonline.com/{{Input Required: Tenant-ID}}/oauth2/v2.0/token"
-        authStyle: InParams
-        clientSecret: "{{Input Required: Client-Secret}}"
+type: "AzureAD-1.0.0"
+adapterConfig: "ewoJInZlcnNpb24iOiAxLAoJImNvbmZpZyI6IHsKICAgICAgICAgICAgICAgICJmaWx0ZXJzIjoge30sCgkJInJlcXVlc3RUaW1lb3V0IjogIjEwcyIsCgkJImF6dXJlYWQiOiB7CiAgICAgICAgICAgICAgICAgICAgImFwaVZlcnNpb24iOiAidjEuMCIKICAgICAgICAgICAgICAgIH0KCX0KfQ=="
```
Comment (shash-sgnl) [inline azure-ad.sgnl.yaml:12]: Do you mind adding the following decoded json as a single-line comment as an example of what it looks like? 
```
{ "version": 1, "config": { "filters": {}, "requestTimeout": "10s",  "azuread": { "apiVersion": "v1.0" } } }
```

File: okta.sgnl.yaml [hunk 2]
```diff
@@ -8,22 +8,18 @@ defaultSyncFrequency: HOURLY
...
-        clientId: "{{Input Required: Client-ID}}"
-        tokenUrl: "https://{{Input Required: Okta-Domain}}/oauth2/v1/token"
-        authStyle: AutoDetect
-        clientSecret: "{{Input Required: Client-Secret}}"
-        scope: "okta.users.read okta.groups.read"
-    - bearer:
-        authToken: "{{Input Required: token}}"
-
+type: "Okta-1.0.0"
+adapterConfig: "ewogICAgInZlcnNpb24iOiAxLAogICAgImNvbmZpZyI6IHsKICAgICAgICAicmVxdWVzdFRpbWVvdXQiOiAiMTBzIiwKICAgICAgICAib2t0YSI6IHsKICAgICAgICAgICAgImFwaVZlcnNpb24iOiAidjEiCiAgICAgICAgfQogICAgfQp9"
```
Comment (shash-sgnl) [inline okta.sgnl.yaml:12]: Please add the following as a single-line comment here as an example of what it looks like:
```
{
    "version": 1,
    "config": {
        "requestTimeout": "10s",
        "okta": {
            "apiVersion": "v1"
        }
    }
}
```

File: salesforce.sgnl.yaml [hunk 3]
```diff
@@ -8,18 +8,15 @@ defaultSyncFrequency: HOURLY
...
-  apiVersion: "V1"
-  disableTls: true
-  auth:
-    - oAuth2ClientCredentials:
-        clientId: "{{Input Required: Client-ID}}"
-        tokenUrl: "https://{{Input Required: Tenant-Name}}.my.salesforce.com/services/oauth2/token?grant_type=client_credentials"
-        authStyle: AutoDetect
-        clientSecret: "{{Input Required: Client-Secret}}"
+type: "Salesforce-1.0.0"
+adapterConfig: "ewoJInZlcnNpb24iOiAxLAoJImNvbmZpZyI6IHsKCQkicmVxdWVzdFRpbWVvdXQiOiAiMTBzIiwKCQkic2FsZXNmb3JjZSI6IHsiYXBpVmVyc2lvbiI6IjU2LjAifQoJfQp9"
```
Comment (shash-sgnl) [inline salesforce.sgnl.yaml:12]: Please add the following as a single-line comment here as an example of what it looks like:
```
{
	"version": 1,
	"config": {
		"requestTimeout": "10s",
		"salesforce": {"apiVersion":"56.0"}
	}
}
```

File: servicenow.sgnl.yaml [hunk 4]
```diff
@@ -7,16 +7,13 @@ defaultSyncFrequency: HOURLY
...
-  port: 80
-  config: "CnsKCSJ2ZXJzaW9uIjogMSwKCSJjb25maWciOiB7CgkJInJlcXVlc3RUaW1lb3V0IjogIjEwcyIsCgkJInNlcnZpY2Vub3ciOiB7fQoJfQp9Cg=="
-  apiVersion: "V1"
-  disableTls: true
-  auth:
-    - basic:
-        username: "{{Input Required: Username}}"
-        password: "{{Input Required: Password}}"
+type: "ServiceNow-1.0.0"
+adapterConfig: "CnsKCSJ2ZXJzaW9uIjogMSwKCSJjb25maWciOiB7CgkJInJlcXVlc3RUaW1lb3V0IjogIjEwcyIsCgkJInNlcnZpY2Vub3ciOiB7fQoJfQp9Cg=="
```
Comment (shash-sgnl) [inline servicenow.sgnl.yaml:11]: Please add the following as a single-line comment here as an example of what it looks like: 
```
{
	"version": 1,
	"config": {
		"requestTimeout": "10s",
		"servicenow": {}
	}
}
```

**Review by shash-sgnl — CHANGES_REQUESTED** [review]

**Review by shash-sgnl — APPROVED** [review]

**Pull request merged**